crates/ruff/src/rules/flake8_pyi/rules/unnecessary_literal_union.rs

```diff
@@ -0,0 +1,115 @@
+use ruff_python_semantic::SemanticModel;
+use rustpython_parser::ast::{self, Expr, Operator, Ranged};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for the presence of multiple literal types in a union.
+///
+/// ## Why is this bad?
+/// Literal types accept multiple arguments and it is clearer to specify them
+/// as a single literal.
+///
+/// ## Example
+/// ```python
+/// field: Literal[1] | Literal[2]
```

Comment:
Nit: maybe `from typing import Literal` on these for completeness in the examples?